frontend/src/tipsyselector/circleselector.tsx

```diff
@@ -0,0 +1,180 @@
+    }>(() => {
+        const knobRadius = (layoutOptions?.knobRadius) ?? 10;
+        const width = layoutOptions.radius * 2 + knobRadius * 2;
+
+        const center = new CartesianPoint(width / 2, width / 2);
+
+        const startAngleDegrees = layoutOptions.startAngle.convertToDegrees();
+        const endAngleDegrees = layoutOptions.endAngle.convertToDegrees();
+
+        const endAngleNormalizedDegrees = new PolarPoint(center, layoutOptions.radius, endAngleDegrees)
+            .normalize(startAngleDegrees)
+            .angle.convertToDegrees();
+
+        const knobPositionPolar = new PolarPoint(
+            center,
+            layoutOptions.radius,
+            new DegreeAngle(endAngleNormalizedDegrees.degrees * value + startAngleDegrees.degrees)
+        );
+        const knobPositionCartesian = knobPositionPolar.convertToCartesian();
+
+        return {
+            width,
+            center,
+            knobRadius,
+            startAngleDegrees,
+            endAngleNormalizedDegrees,
+            knobPositionPolar,
+            knobPositionCartesian
+        }
+    }, [value, layoutOptions]);
+
+    const pressed = React.useRef<{ pressed: boolean, lastRegisteredInput: number }>({
+        pressed: false,
+        lastRegisteredInput: 0
+    });
+    function onPress() {
+        pressed.current.pressed = true;
+    }
+    function onRelease() {
+        pressed.current.pressed = false;
+        pressed.current.lastRegisteredInput = 0;
+    }
+
+    function onMove(event: React.MouseEvent<HTMLDivElement>) {
+        /* Debounce and check for mouse press */
+        const currentTime = new Date().getTime();
+        // Mouse movement only, so the input must not change the value
+        if (!pressed.current.pressed || currentTime - 25 < pressed.current.lastRegisteredInput) {
+            return;
+        }
+
+        /* Determine point on circle closest to the cursor */
+        function calculateClosestPointOnCircle(point: CartesianPoint): { point: CartesianPoint, offset: number } {
+            const centerToPoint: Vector = new CartesianPoint(
+                point.x - center.x,
+                point.y - center.y
+            );
+
+            const centerToPointLength = centerToPoint.calculateEuclideanDistance();
+
+            return {
+                offset: Math.abs(layoutOptions.radius - centerToPointLength),
+                point: new CartesianPoint(
+                    centerToPoint.x / centerToPointLength * layoutOptions.radius + center.x,
+                    centerToPoint.y / centerToPointLength * layoutOptions.radius + center.y
+                )
+            };
+        }
+
+        // Rectangle containing all elements. Offsets are relative to the viewport.
+        const targetDimensionsInViewport: DOMRect = event.currentTarget.getBoundingClientRect();
+        const cursorPosition: Vector = new CartesianPoint(
+            event.clientX - targetDimensionsInViewport.left,
```

Comment:
@Quaffel Ah, missed this: the [TouchEvent](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent) does not have a clientX and clientY. Instead you need to access `.touches[0]`, see [here](https://developer.mozilla.org/en-US/docs/Web/API/Touch/clientX)